### PR TITLE
[BIOMAGE-1568] - Refactor Plot Container

### PR DESCRIPTION
**Pull request by aerlaut**
# Description
#### Problem
Currently, the "reset" button for plots is located in the header beside the log-out button. This is not good UI as it is confusing for the user - is the "reset" associated with the experiment? or the plot? This PR solves that issue.

#### Solution
- Move the reset button from the plot header into the plot container, so it's clearer that "Reset" means the plot.
- Remove the ability of the plot to be minimized.
- Better keys of plot styling components so some options are open by default.

### Staging link
https://ui-agi-stealthy-indri.scp-staging.biomage.net/

# Details
#### URL to issue
https://biomage.atlassian.net/browse/BIOMAGE-1568

#### Link to staging deployment URL (or set N/A)
N/A

#### Integration test branch
master

# Merge checklist
Your changes will be ready for merging after all of the steps below have been completed.

### Code updates
Have best practices and ongoing refactors being observed in this PR
- [ ] Migrated any selector / reducer used to the new format.

### Manual/unit testing
- [X] Tested changes using InfraMock locally **or** no tests required for change, e.g. Kubernetes chart updates.
- [X] Validated that current unit tests for code work as expected and are sufficient for code coverage **or** no unit tests required for change, e.g. documentation update.
- [X] Unit tests written **or** no unit tests required for change, e.g. documentation update.

### Integration testing
**You must check the box below** to run integration tests on the latest commit on your PR branch.
Integration tests have to pass before the PR can be merged. Without checking the box, your PR
**will not pass** the required status checks for merging.

- [ ] Started end-to-end tests on the latest commit.

### Documentation updates
- [X] Relevant Github READMEs updated **or** no GitHub README updates required.
- [X] Relevant Wiki pages created/updated **or** no Wiki updates required.

### Optional
- [ ] Staging environment is unstaged before merging.
- [ ] Photo of a cute animal attached to this PR.

### CODE REVIEW

File: src/components/plots/styling/PlotStyling.jsx [hunk 1]
```diff
@@ -27,9 +27,12 @@ import VolcanoMarkersEditor from './volcano/MarkersEditor';
 import VolcanoDisplayLabels from './volcano/DisplayLabels';
 
 const { Panel } = Collapse;
+
+const formatPanelKey = (key) => key.trim().toLowerCase().replace(' ', '-');
```
Comment (cosa65) [inline src/components/plots/styling/PlotStyling.jsx]: Remove this one and use `_.kebabCase`

File: src/redux/actions/componentConfig/resetPlotConfig.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+      },
+    );
+
+    dispatch({
+      type: RESET_CONFIG,
+      payload: {
+        plotUuid,
+        config: defaultConfig,
+      },
+    });
```
Comment (cosa65) [inline src/redux/actions/componentConfig/resetPlotConfig.js:28]: Just use `RESET_CONFIG` and add what you want of `SAVE_CONFIG` in the reducer for this

File: src/redux/reducers/componentConfig/saveConfig.js [hunk 3]
```diff
@@ -1,13 +1,16 @@
...
-  };
-};
+/* eslint-disable no-param-reassign */
+import produce, { setAutoFreeze } from 'immer';
+
+import initialState from 'redux/reducers/componentConfig/initialState';
+
+// Vega modifies the plot config object, so we need to disable Immer autofreeze.
+// If we don't do this, the reducer will freeze the object and the plot will not update.
+setAutoFreeze(false);
```
Comment (cosa65) [inline src/redux/reducers/componentConfig/saveConfig.js]: Try removing this and using clone to deal iwth the vega behavior

File: src/components/plots/styling/PlotStyling.jsx (hunk 1 above)
Comment (aerlaut) [reply src/components/plots/styling/PlotStyling.jsx]: changed

File: src/redux/reducers/componentConfig/saveConfig.js (hunk 3 above)
Comment (aerlaut) [reply src/redux/reducers/componentConfig/saveConfig.js]: removed

**Review by cosa65 — APPROVED** [review]

**Review by cosa65 — APPROVED** [review]

**Pull request merged**